lib/posix/options/pthread.c

```diff
@@ -1092,8 +1088,13 @@ int pthread_join(pthread_t pthread, void **status)
 		break;
 	}
 
-	ret = k_thread_join(&t->thread, K_FOREVER);
-	/* other possibilities? */
+	ret = k_thread_join(&t->thread, timeout);
+	if (ret == -EBUSY) {
+		return EBUSY;
+	} else if (ret == -EAGAIN) {
+		return ETIMEDOUT;
+	}
+	/* Can only be ok or -EDEADLK, which should never occur for pthreads */
 	__ASSERT_NO_MSG(ret == 0);
```

Comment:
```suggestion
	if (ret == -EAGAIN) {
		return ETIMEDOUT;
	} else if (ret < 0) {
		/* since this is a non-portable API, we can also return EBUSY or EDEADLK */
		return -ret;
	}
```